### PR TITLE
For import fix, support path mapping value that begins in "./" or ends in ".ts"

### LINKED ISSUE #20525: Add path alias to the auto import suggestions
_From @Maus3rVonDutch on December 2, 2017 9:52_

Right now the auto import feature provides the file location of any symbol, which is great but could be improved upon. In the ts compiler options you have the option to provide path aliases, so you can combine multiple symbols in one import declaration:

tsconfig.json
```
{
    "compilerOptions": {
        ...
        "paths": {
            "@my-application/common": [
                "./src/app/common/index"
            ],
            "@my-application/module1": [
                "./src/app/module1/index"
            ],
            "@my-application/module2": [
                "./src/app/module2/index"
            ]          
        }
}
```

With this configuration you can import multiple classes in one declaration:
```
import { CommonA, CommonB } from '@my-application/common';
import { ClassA, ClassB, ClassC } from '@my-application/module1';
import { ClassD, ClassE } from '@my-application/module2';
```
It would be a great feature if the import suggestion provider would be able to pick up those paths from the tsconfig.json file and provide those aliases as well. Maybe the provider could have a configuration that allows you to include both options (the local file path and the alias) or just one or the other. As a note, when choosing the alias, the import feature would have to add the symbol to any existing declaration (and not just always add the alias as a new import line)

_Copied from original issue: Microsoft/vscode#39459_

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #20525

### CODE REVIEW

File: src/services/codefixes/importFixes.ts [hunk 1]
```diff
@@ -432,22 +432,30 @@ namespace ts.codefix {
...
+                        return key;
+                    }
                 }
             }
         }
     }
 
+    /** Convert "./x" (and "././x") to "x". */
+    function removeLeadingDotSlash(path: string): string {
+        return startsWith(path, "./") || startsWith(path, ".\\") ? removeLeadingDotSlash(path.slice(2)) : path;
```
Comment (mhegazy) [inline src/services/codefixes/importFixes.ts]: the two relative paths are not relative to the same thing.. the input is relative to the current file, but the path is relative to the tsconfig.json. do not think removing the `./` is the right thing to do here. you probably want to expand both, or normalize one in terms of the other if it starts with a `./` or a `../`

Comment (andy-ms) [reply src/services/codefixes/importFixes.ts]: The two parameters to `tryGetModuleNameFromPaths` are relative to the baseUrl. I pushed a commit that updates their names.

Comment (mhegazy) [reply src/services/codefixes/importFixes.ts]: ahh.. good to know :)

what about  `..\`?

Comment (andy-ms) [reply src/services/codefixes/importFixes.ts]: Good catch, that needed a fix too.

Comment (mhegazy) [reply src/services/codefixes/importFixes.ts]: can u use getNormalizedPathComponents instead?

Comment (andy-ms) [reply src/services/codefixes/importFixes.ts]: Actually, this function is unused! #21092

**Pull request merged**